src/Test/Utilities/Portable/FX/ProcessUtilities.cs

```diff
@@ -43,6 +46,46 @@ public static class ProcessUtilities
             {
                 startInfo.CreateNoWindow = true;
             }
+After:
+            if (fileName == null)
+                throw new ArgumentNullException(nameof(fileName));
+
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = fileName,
+                Arguments = arguments,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                RedirectStandardInput = stdInput != null || redirectStandardInput,
+                WorkingDirectory = workingDirectory
+            };
+
+            // In case the process is a console application that expects standard input
+            // do not set CreateNoWindow to true to ensure that the input encoding
+            // of both the test and the process fileName is equal.
+*/
+            if (fileName == null)
+                throw new ArgumentNullException(nameof(fileName));
+
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = fileName,
+                Arguments = arguments,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                RedirectStandardInput = stdInput != null || redirectStandardInput,
+                WorkingDirectory = workingDirectory
+            };
+
+            // In case the process is a console application that expects standard input
+            // do not set CreateNoWindow to true to ensure that the input encoding
+            // of both the test and the process fileName is equal.
+            if (stdInput == null)
+            {
+                startInfo.CreateNoWindow = true;
+            }
```

Comment:
@sharwell  something with your tool?

Reply:
It's a bug in the workspaces implementation or in the BatchFixAllProvider. Revert this file, then re-run the Fix All. Sometimes it takes a few tries to not have any of these cases.